Add task-arn and container-name labels.

ejholmes opened the pull request:
This is in regards to https://github.com/aws/amazon-ecs-agent/issues. This adds the following labels to the docker container when it is run:

* `com.amazonaws.ecs.task-arn`: The task ARN associated with this container.
* `com.amazonaws.ecs.container-name`: The name of the container, from the task definition.

I've left out adding a `com.amazonaws.ecs.task-definition-arn` label because it seems like that information [isn't made available by the acs client](https://github.com/aws/amazon-ecs-agent/blob/a2fe6ba7e1dd58bcd518b8e91492f4630c067470/agent/acs/model/ecsacs/api.go#L272-L290).

Closes https://github.com/aws/amazon-ecs-agent/issues/82

/cc @samuelkarp

samuelkarp commented on agent/api/task_test.go:
```diff
@@ -96,6 +96,28 @@ func TestDockerHostConfigVolumesFrom(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(config.Labels, map[string]string{
```
Might want to move the inline map you have out to a named variable just for readability.

samuelkarp commented:
You're right about the Task Definition ARN.  How about adding `com.amazonaws.ecs.task-definition-family` and `com.amazonaws.ecs.task-definition-version` instead?  Those two will have most of the interesting components of the ARN anyway.

ejholmes commented:
Updated to add the `com.amazonaws.ecs.task-definition-family` and `com.amazonaws.ecs.task-definition-version` labels and moved the expected map into a named variable.

samuelkarp commented:
Missed this before: can you re-create the pull request against our `dev` branch?  Otherwise the code looks great!

ejholmes commented:
Ah dang. I swear I read the contributing guidelines :). Will do!
